packages/contentstack-import/src/import/modules/entries.ts

```diff
@@ -402,6 +449,135 @@ export default class EntriesImport extends BaseClass {
+      basePath: path.join(this.entriesMapperPath, cTUid, locale),
+      chunkFileSize: this.entriesConfig.chunkFileSize,
+      keepMetadata: false,
+      useIndexer: true,
+      omitKeys: this.entriesConfig.invalidKeys,
+    });
+    // log(this.importConfig, `Starting to update entries with references for ${cTUid} in locale ${locale}`, 'info');
+
+    const contentType = find(this.cTs, { uid: cTUid });
+
+    const onSuccess = ({ response, apiData: entry, additionalInfo }: any) => {
+      log(this.importConfig, `Replaced entry: '${entry.title}' of content type ${cTUid} in locale ${locale}`, 'info');
+      this.entriesUidMapper[entry.uid] = response.uid;
+      entriesReplaceFileHelper.writeIntoFile({ [entry.uid]: entry } as any, { mapKeyVal: true });
+    };
+    const onReject = ({ error, apiData: { uid, title } }: any) => {
+      log(this.importConfig, `${title} entry of content type ${cTUid} in locale ${locale} failed to replace`, 'error');
+      log(this.importConfig, formatError(error), 'error');
+      this.failedEntries.push({
+        content_type: cTUid,
+        locale,
+        entry: { uid: this.entriesUidMapper[uid], title },
+        entryId: uid,
+      });
+    };
+
+    for (const index in indexer) {
+      const chunk = await fs.readChunkFiles.next().catch((error) => {
+        log(this.importConfig, formatError(error), 'error');
+      });
+
+      if (chunk) {
+        let apiContent = values(chunk as Record<string, any>[]);
+        await this.makeConcurrentCall(
+          {
+            apiContent,
+            processName,
+            indexerCount,
+            currentIndexer: +index,
+            apiParams: {
+              reject: onReject,
+              resolve: onSuccess,
+              entity: 'update-entries',
+              includeParamOnCompletion: true,
+              additionalInfo: { contentType, locale, cTUid },
+            },
+            concurrencyLimit: this.importConcurrency,
+          },
+          this.replaceEntriesHandler.bind(this),
+        ).then(() => {
+          entriesReplaceFileHelper?.completeFile(true);
+          log(this.importConfig, `Replaced entries for content type ${cTUid} in locale ${locale}`, 'success');
+        });
+      }
+    }
+  }
+
+  async replaceEntriesHandler({
+    apiParams,
+    element: entry,
+    isLastRequest,
+  }: {
+    apiParams: ApiOptions;
+    element: Record<string, string>;
+    isLastRequest: boolean;
+  }) {
+    const { additionalInfo: { cTUid, locale } = {} } = apiParams;
+    return new Promise(async (resolve, reject) => {
+      const { items: [entryInStack] = [] }: any =
+        (await this.stack
+          .contentType(cTUid)
+          .entry()
+          .query({ query: { title: entry.title, locale } })
+          .findOne()
+          .catch((error: Error) => {
+            apiParams.reject({
+              error,
+              apiData: entry,
+            });
+            reject(true);
+          })) || {};
```

Comment:
[MINOR] Can we take this block of code out of Promise? so that we do need not to use async inside the promise which will throw a lint issue.

Reply:
@antonyagustine , this particular code is disabled using a overwrite flag, changing structure would need more testing. will consider this point on the phase 2 when entries will be enabled. For now not getting any lint issues